cmd/talosctl/cmd/mgmt/cluster/create.go

```diff
@@ -176,6 +176,11 @@ func create(ctx context.Context) (err error) {
 			generate.WithDebug(configDebug),
 		}
 
+		// Do not persist config files when using docker-based provisioner
+		if provisionerName == "docker" {
```

Comment:
you can use call `provisioner.GenOptions()` call below to push `generate.WithPersist(false)` from docker provisioner without doing special case here